src/translate.py

```diff
@@ -1120,11 +1140,15 @@ class StructAccess(Expression):
     offset: int
     target_size: Optional[int]
     field_path: Optional[AccessPath] = field(compare=False)
-    stack_info: StackInfo = field(compare=False, repr=False)
+    stack_info: Optional[StackInfo] = field(compare=False, repr=False)
```

Comment:
(this is pretty ugly, we should probably make this a thread-local at some point...)

Reply:
For fun, I tried putting `has_nonzero_access` in `TypeData`, so that `record_struct_access()` & `has_nonzero_access()` are done on `Type`s instead of using `StackInfo` & `Expression`s. 

As expected, it changed a lot of `*x` into `x->unk_0`, but maybe some of these would be better written as `x[0]`? There's also some changes around `GlobalSymbol`s, those may need additional work too? [Full diff on OOT+MM](https://gist.github.com/zbanks/19923b394923b1b11083e1f4338a292c).